controllers/version_update.go

```diff
@@ -406,7 +419,7 @@ func parseImageString(imageString string) (*Version, error) {
 }
 
 // compare two parsed versions
-// -1: left < right
+// -n: left < right, nth component differs (1-indexed)
 // 0: left = right
 // 1: left > right
 func compareVersion(l, r *Version) int {
```

Comment:
`compareVersion()` can be simplified to similarly return 1,2,3,4 when left > right. Moreover, you won't need to run 3 for loops. The result can be obtained in one for loop.

The caller doesn't need to use `switch case`. It can instead use `if..else if...else` construct.